python/nano/src/bigdl/nano/pytorch/trainer/Trainer.py

```diff
@@ -156,3 +154,68 @@ def compile(model: nn.Module,
+                            onnxrt_integer, onnxrt_qlinear or mxnet; allow new framework backend
+                            extension. Default: pytorch_fx. Consistent with Intel Neural Compressor
+                            Quantization.
+
+        :param approach:    static or dynamic.
+                            static: post_training_static_quant,
+                            dynamic: post_training_dynamic_quant,
+                            Default: static.
+        :param tuning_strategy:    bayesian, basic, mse, sigopt. Default: bayesian.
+        :param accuracy_criterion:  Tolerable accuracy drop.
+                                    accuracy_criterion = {'relative': 0.1, higher_is_better=True}
+                                    allows relative accuracy loss: 1%. accuracy_criterion =
+                                    {'absolute': 0.99, higher_is_better=Flase} means accuracy < 0.99
+                                     must be satisfied.
+        :param timeout:     Tuning timeout (seconds). Default: 0,  which means early stop.
+                            Combine with max_trials field to decide when to exit.
+        :param max_trials:  Max tune times. Default: 1.
+                            Combine with timeout field to decide when to exit.
+                            "timeout=0, max_trials=1" means it will try quantization only once and
+                            return satisfying best model.
+        :return:             A GraphModule. If there is no model found, return None.:
+        """
+        if backend == 'inc':
+            from bigdl.nano.quantization import QuantizationINC
+            from neural_compressor.experimental import common
+            approach_map = {
+                'static': 'post_training_static_quant',
+                'dynamic': 'post_training_dynamic_quant'
+            }
+            approach = approach_map.get(approach)
+            quantizer = QuantizationINC(framework=framework, conf=conf, approach=approach,
+                                        tuning_strategy=tuning_strategy,
+                                        accuracy_criterion=accuracy_criterion,
+                                        timeout=timeout, max_trials=max_trials)
+            if isinstance(pl_model, LightningModuleFromTorch):
+                # LightningModuleFromTorch.forward fails to trace in FX, so replace it temporarily
+                quantizer.model = pl_model.model
+            else:
+                quantizer.model = pl_model
+
+            if val_dataloader and metric:
+                from bigdl.nano.quantization.quantization_inc import TorchMetricForINC
+                quantizer.eval_dataloader = val_dataloader
+                quantizer.metric = common.Metric(TorchMetricForINC, metric=metric)
+            quantizer.calib_dataloader = calib_dataloader
+            quantized = quantizer()
+            return quantized.model if quantized else None
```

Comment:
In what situation does `quantized` evaluate to `False`? Maybe raising an error to indicate the quantization is failed?

Reply:
If no satisfied model found, `quantized` would be `None`. For example, users want a quantized model within 1% acc drop while INC can't find such model with current config, then it'll return `None`. And INC will warn that so we don't need to catch such warning.